tests/test_examples.py

```diff
@@ -184,11 +180,22 @@ def test_dglobal_caching(method_inst):
     assert method._global.cache_info().misses == 1
     assert method._global.cache_info().currsize == 1
 
+@pytest.mark.filterwarnings('ignore::sklearn.exceptions.ConvergenceWarning')
+@pytest.mark.parametrize("method_inst", ["kalman", "trend_filtered", "spectral"], indirect=True)
+def test_global_caching_dd(method_inst):
+    # make sure we're not recomputing expensive _global() method
+    x, t, method = method_inst
+    method.d(x, t)
+    method.d(x, t)
```

Comment:
Because `spectral` doesn't actually have any separate x smoothing, calling `x` followed by `d` didn't really make any sense. So I've put a test to call `d`, then `d` again and make sure the cache hits are still only 1. I'd like to combine this test with the one above, but calling `x`, `d`, `d` results in 2 calls to `_global`. Not sure why. Likewise, calling `d`, `d`, `d` results in 2 hits, and `d`, `d`, `d`, `d` results in 3! I don't know enough about caching to know why this is.

Reply:
My first comment kinda explains this.  TL;DR: `method.x` does not call `method._global`, it calls `method.compute_x`.  Only the methods that actually smooth `x` then call `_global`.

You don't need to test cacheing for `spectral`.